samples/drivers/watchdog/src/iwdg_main.c

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2017 RnDity Sp. z o.o.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <watchdog.h>
+#include <misc/printk.h>
+
+/*
+ * The independent watchdog (IWDG) is clocked by its own dedicated
+ * 40kHz low-speed internal clock (LSI).
+ *
+ * Prescaler register should be set to 'divider /16' and the reload
+ * register to 2500. Then it takes 1 second a reset signal is generated.
+ */
+
+void main(void)
+{
+	struct device *iwdg = device_get_binding("IWDG");
+	wdt_enable(iwdg);
+
+	struct k_timer iwdg_timer;
+	k_timer_init(&iwdg_timer, NULL, NULL);
+
+	uint8_t counter = 0;
+
+	while (1) {
+		printk("Welcome...\n");
+
+		if (counter % 2 == 0) {
+			/* Less than IWDG reload value. */
+			k_timer_start(&iwdg_timer, K_MSEC(900), 0);
+		} else {
+			/* More than IWDG reload value. */
+			k_timer_start(&iwdg_timer, K_MSEC(1100), 0);
+		}
+
+		counter++;
+		/* Wait till application timer expires. */
+		k_timer_status_sync(&iwdg_timer);
+
+		printk("Watchdog refreshed...\n");
+		wdt_reload(iwdg);
```

Comment:
I tested on nucleo_f334ro and had to add a k_sleep(2) otherwise message was never printed.
With this delay, I get:
Welcome...                                                                      
Welcome...                                                                      
Watchdog refreshed...                                                           
Welcome...            

I think it is as expected